frontend/src/components/table/useCheckboxTableBase.ts

```diff
@@ -17,6 +17,7 @@ const useCheckboxTableBase = <T>(
   setSelectedData: React.Dispatch<React.SetStateAction<T[]>>,
   dataMappingHelper: (selectData: T) => string,
   selectAll?: { selected?: boolean; disabled?: boolean },
+  persistSelections?: boolean,
```

Comment:
code quality wise we should put these two into a `opts` param so we can avoid putting undefined in when we dont need to set selectAll

Reply:
I disagree that this suggestion improves code quality, and this seems more like a personal preference, but have updated anyways.

Reply:
My reasoning is that as function params start to have multiple optional params you can start to have things like this in an extreme case:

```tsx
const someFunc = (p1: string, p2: string, op1?: string, op2?: string, op3?: string, op4?: string) => {...}

someFunc("one", "two", undefined, undefined, undefined, "optional")
```

With one optional param its not a big deal, two sure, but three definitely want to do something about it. The reason I  flagged it is becuase it was talked about at some point prior as something we wanted to shy away from.

Reply:
There was a jira to look to add a linter rule for it, but i dont think it was added as one doesn't exist

https://issues.redhat.com/browse/RHOAIENG-513